DataFormats/SiStripCluster/src/SiStripCluster.cc

```diff
@@ -28,7 +28,7 @@ SiStripCluster::SiStripCluster(const SiStripApproximateCluster cluster) : error_
   amplitudes_.resize(cluster.width(), cluster.avgCharge());
 
   //initialize firstStrip_
-  firstStrip_ = cluster.barycenter() - cluster.width() / 2;
+  firstStrip_ = std::max(0.f, cluster.barycenter() - cluster.width() / 2);
```

Comment:
sure. I have more doubts though:

1) given `cluster.width()` here is a `uint8_t`  https://github.com/cms-sw/cmssw/blob/9e6c192dd06b019f5bb791a42df0a6e8acd389c4/DataFormats/SiStripCluster/interface/SiStripApproximateCluster.h#L25 should it be divided by `2.f` to avoid truncating the half-width of half-strip?

2) I am not sure if there's a better (more efficient) way of computing the max since this should be a pretty intensive call in production.

@abaty @mandrenguyen

Reply:
Hi,

I think Marco is correct on item 1, we should be dividing by a float here.  I am not familiar with the most efficient way of implementing item 2, although we could check the timing if it is a strong concern.

Reply:
plase ```0.5f*float(cluster.width())```